ri_public_examples/download_files.py

```diff
@@ -10,21 +10,28 @@
 DEFAULT_CONFIG_PATH = "configs/stress_test_config.json"
 
 
-def _download_file(remote_path: str, local_path: Path) -> None:
+def _download_file(remote_path: str, local_path: Path) -> int:
     """Downloads file to local path."""
     if not local_path.parent.exists():
         local_path.parent.mkdir(parents=True)
     r = requests.get(remote_path)
-    with open(local_path, 'wb') as f:
-        f.write(r.content)
+    if r.status_code == 200:
+        with open(local_path, 'wb') as f:
+            f.write(r.content)
+    
+    return r.status_code
 
-def _fetch_urls_and_outpaths(remote_file_path: str, out_dir: Path) -> List[Tuple[str, Path]]:
+def _fetch_urls_and_outpaths(
+    remote_file_path: str, out_dir: Path
+) -> Optional[List[Tuple[str, Path]]]:
     """Fetch urls and output paths from text file."""
     # TODO: add data_urls/configs to package data directly
     # first download remote file path
     local_path = out_dir / Path(remote_file_path).name
     
-    _download_file(remote_file_path, local_path)
+    status_code = _download_file(remote_file_path, local_path)
+    if status_code != 200:
+        return None
```

Comment:
Should we raise some warning to the user if the download attempt fails?

Reply:
Thumbs up as in we should, or thumbs up as in this is already fine and better as-is?

Reply:
i added a warning in download_file above